src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -225,6 +226,10 @@ protected function refreshUser(TokenInterface $token)
                 $this->logger->debug('Token was deauthenticated after trying to refresh it.');
             }
 
+            if (null !== $this->dispatcher) {
+                $this->dispatcher->dispatch(new DeauthenticatedEvent($token, $newToken), DeauthenticatedEvent::class);
```

Comment:
@xabbuh unless I'm mistaken it does not because `$token->setUser()` has not been called at this point due to the `continue` statement l190. Test seem to prove it https://github.com/symfony/symfony/pull/31138/files/40e42183b841a0d713499a1a867a67bc4b1c2446#diff-65644157b3a60649add913034d597198R335

Reply:
You are right. I missed that.